src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs

```diff
@@ -657,11 +658,15 @@ public async Task<Schema> GetSchemaAsync(int id, string format = null)
         /// <inheritdoc/>
         public async Task<Schema> GetSchemaBySubjectAndIdAsync(string subject, int id, string format = null)
         {
+            if (this.schemaById.TryGetValue(id, out Schema schema) && !checkSchemaMatchesFormat(format, schema.SchemaString))
```

Comment:
I’m wondering if we could eliminate the `checkSchemaMatchesFormat` method. In my workaround, I created a cache that uses all three parameters as the key, which allowed me to skip this check. One idea that comes to mind is extending the `Schema` object with an additional property to store the format information.  

@rayokota, what are your thoughts on this approach?

Reply:
.NET 8 introduces a really nice method that allows you to check if a string is valid Base64 encoding without any allocations, making this check a no-op in terms of performance.

Reference: [Base64.IsValid](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.text.base64.isvalid?view=net-8.0)

We could introduce a helper method that provides a performance boost for .NET 8 and onward compilations while either falling back to the old mechanism for older .NET versions or offering a polyfill.

Here’s the code snippet:

```csharp
internal static bool IsBase64String(string value)
{
#if NET8_0_OR_GREATER
    return Base64.IsValid(value);
#else
    try
    {
        _ = Convert.FromBase64String(value);
        return true;
    }
    catch (FormatException)
    {
        return false;
    }
#endif
}
```